kong/clustering/init.lua

```diff
@@ -1,122 +1,22 @@
 local openssl_x509 = require("resty.openssl.x509")
-local isempty = require("table.isempty")
-local isarray = require("table.isarray")
-local nkeys = require("table.nkeys")
-local new_tab = require("table.new")
 local ngx_log = ngx.log
-local ngx_null = ngx.null
-local ngx_md5 = ngx.md5
-local ngx_md5_bin = ngx.md5_bin
-local tostring = tostring
 local assert = assert
-local error = error
-local concat = table.concat
-local pairs = pairs
 local sort = table.sort
-local type = type
 
 local ngx_ERR = ngx.ERR
 local ngx_DEBUG = ngx.DEBUG
 
-local DECLARATIVE_EMPTY_CONFIG_HASH = constants.DECLARATIVE_EMPTY_CONFIG_HASH
 local _log_prefix = "[clustering] "
 
-
-local MT = { __index = _M, }
-
-
-local function to_sorted_string(value)
-  if value == ngx_null then
-    return "/null/"
-  end
-
-  local t = type(value)
-  if t == "string" or t == "number" then
-    return value
-
-  elseif t == "boolean" then
-    return tostring(value)
-
-  elseif t == "table" then
-    if isempty(value) then
-      return "{}"
-
-    elseif isarray(value) then
-      local count = #value
-      if count == 1 then
-        return to_sorted_string(value[1])
-
-      elseif count == 2 then
-        return to_sorted_string(value[1]) .. ";" ..
-               to_sorted_string(value[2])
-
-      elseif count == 3 then
-        return to_sorted_string(value[1]) .. ";" ..
-               to_sorted_string(value[2]) .. ";" ..
-               to_sorted_string(value[3])
-
-      elseif count == 4 then
-        return to_sorted_string(value[1]) .. ";" ..
-               to_sorted_string(value[2]) .. ";" ..
-               to_sorted_string(value[3]) .. ";" ..
-               to_sorted_string(value[4])
-
-      elseif count == 5 then
-        return to_sorted_string(value[1]) .. ";" ..
-               to_sorted_string(value[2]) .. ";" ..
-               to_sorted_string(value[3]) .. ";" ..
-               to_sorted_string(value[4]) .. ";" ..
-               to_sorted_string(value[5])
-      end
-
-      local i = 0
-      local o = new_tab(count < 100 and count or 100, 0)
-      for j = 1, count do
-        i = i + 1
-        o[i] = to_sorted_string(value[j])
-
-        if j % 100 == 0 then
-          i = 1
-          o[i] = ngx_md5_bin(concat(o, ";", 1, 100))
-        end
-      end
-
-      return ngx_md5_bin(concat(o, ";", 1, i))
-
-    else
-      local count = nkeys(value)
-      local keys = new_tab(count, 0)
-      local i = 0
-      for k in pairs(value) do
-        i = i + 1
-        keys[i] = k
-      end
-
-      sort(keys)
-
-      local o = new_tab(count, 0)
-      for i = 1, count do
-        o[i] = keys[i] .. ":" .. to_sorted_string(value[keys[i]])
-      end
-
-      value = concat(o, ";", 1, count)
-
-      return #value > 512 and ngx_md5_bin(value) or value
-    end
-
-  else
-    error("invalid type to be sorted (JSON types are supported)")
-  end
-end
+local check_protocol_support =
+  require("kong.clustering.utils").check_protocol_support
```

Comment:
Unless necessary, please put the `require`s at the top of the file. Seeing all the file dependencies in a single place helps when reviewing.

Reply:
done